### ISSUE TITLE
Upload sourcemaps issue with latest 2.4.0 CLI release

### ISSUE THREAD

**Issue by spra85** [getsentry/sentry-cli#1283]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?

Javascript Sentry CLI, latest - 2.4.0

### Steps to Reproduce

Basic command in our CI build is below, where `REACT_APP_COMMIT` is set to essentially Git SHA for release

```bash
sentry-cli releases files $REACT_APP_COMMIT upload-sourcemaps build/static/js
```

Our installation of Sentry-CLI used this command from the README: 

```
curl -sL https://sentry.io/get-cli/ | bash
```

With the most recent version bump, the command above started failing with "release not found".

Pinning the version to the one right before does work, which implies a 2.4.0 change caused this.

```
curl -sL https://sentry.io/get-cli/ | SENTRY_CLI_VERSION=2.3.1 bash
```

It's possible we weren't correctly using it in the first place and 2.4.0 is more strict; for instance, from the docs it suggests this - 

```bash
#!/bin/sh
sentry-cli releases new "$VERSION"
# do your build steps here
# once you are done, finalize
sentry-cli releases finalize "$VERSION"
```

With 2.4.0, does the CLI expect that we would first need to create a new release, prior to trying to upload the source map whereas previous versions allowed that to happen in a single step/command?

### Expected Result

The source map would upload

### Actual Result

```
[Container] 2022/07/13 14:56:32 Running command sentry-cli releases files $REACT_APP_COMMIT upload-sourcemaps build/static/js --validate --url-prefix '~/static/js'
--
error: release not found
```

**Comment by kamilogorek** [comment]
Thanks for the report, should be fixed in #1284, will release a patched version once it's merged.

**Comment by spra85** [comment]
Thanks @kamilogorek, once a patch is available we can retest.  Obviously we have a workaround for now with pinning the version

**Comment by kamilogorek** [comment]
> With 2.4.0, does the CLI expect that we would first need to create a new release, prior to trying to upload the source map whereas previous versions allowed that to happen in a single step/command?

It should do that from the very beginning, but it somehow snuck into the `upload` behavior and stayed that way forever. My feature addition did not intend to change that, so I'll fix that :)

**Comment by joer14** [comment]
Yah this issue blocked a release for us and helped uncover an unpinned dependency. Thanks for reporting @spra85!!!

**Comment by kamilogorek** [comment]
The issue should be fixed in `2.4.1`.
